### PR TITLE
encoder/decoder pb creation

### PR DESCRIPTION
**Pull request by cookingbear**

### CODE REVIEW

**Review by tjadamlee — APPROVED** [review]

File: athena/main.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def build_model_from_jsonfile(jsonfile, pre_run=True):
         model=model,
         optimizer=optimizer,
     )
-    if pre_run or p.pretrained_model is not None:
+    if pre_run:
```
Comment (Some-random) [inline athena/main.py:97]: It should be "if pre_run and p.pretrained_model is not None"

File: athena/deploy_main.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    if hasattr(model, "deploy_encoder") and model.deploy_encoder is not None:
+        frozen_graph = freeze_session(tf1.keras.backend.get_session(),
+                                      output_names=[out.op.name for out in model.deploy_encoder.outputs])
+        tf1.train.write_graph(frozen_graph, deploy_path, "encoder.pb", as_text=False)
+        logging.info("successful save encoder as pb")
+    if hasattr(model, "deploy_decoder") and model.deploy_decoder is not None:
+        frozen_graph = freeze_session(tf1.keras.backend.get_session(),
+                                      output_names=[out.op.name for out in model.deploy_decoder.outputs])
+        tf1.train.write_graph(frozen_graph, deploy_path, "decoder.pb", as_text=False)
+        logging.info("successful save decoder as pb")
```
Comment (Some-random) [inline athena/deploy_main.py:62]: Successfully saved

File: athena/models/speech_transformer.py [hunk 3]
```diff
@@ -259,6 +267,35 @@ def restore_from_pretrained_model(self, pretrained_model, model_type=""):
         else:
             raise ValueError("NOT SUPPORTED")
 
+    def deploy(self):
```
Comment (Some-random) [inline athena/models/speech_transformer.py:270]: We need some comments for this function, something like "frozen graph creation in tf 2.0 requires keras model, so we have to build it again"

File: athena/models/speech_transformer.py [hunk 4]
```diff
@@ -123,6 +128,9 @@ def __init__(self, data_descriptions, config=None):
         # some temp function
         self.random_num = tf.random_uniform_initializer(0, 1)
 
+        self.deploy_encoder = None
```
Comment (Some-random) [inline athena/models/speech_transformer.py:131]: We should also create a 'encoder+decoder' pb. Creating deploy demo will be easier with a single pb.

**Pull request merged**